user/mongoStoreClient.go

```diff
@@ -125,6 +122,22 @@ func (d MongoStoreClient) FindUsers(user *User) (results []*User, err error) {
 	return results, nil
 }
 
+func (d MongoStoreClient) FindUsersByRole(role string) (results []*User, err error) {
+	cpy := d.session.Copy()
+	defer cpy.Close()
+
+	if err = mgoUsersCollection(cpy).Find(bson.M{"roles": role}).All(&results); err != nil {
```

Comment:
so its no longer looking for the role e.g. `clinic` being included in the `roles`?? maybe its just how I am reading it but when doing the match above  that is now deleted we have `bson.M{"roles": bson.M{"$in": user.Roles}}`